src/async-computed.ts

```diff
@@ -0,0 +1,206 @@
+   * status is 'pending', the `value` and `error` properties will be the result
+   * of the previous run of the compute function.
+   *
+   * The status will be 'complete' when the compute function has completed
+   * successfully. If the status is 'complete', the `value` property will be the
+   * result of the previous run of the compute function and the `error` property
+   * will be `undefined`.
+   *
+   * The status will be 'error' when the compute function has completed with an
+   * error. If the status is 'error', the `error` property will be the error
+   * that was thrown by the previous run of the compute function and the `value`
+   * property will be `undefined`.
+   *
+   * This value is read from a signal, so any signals that read it will be
+   * tracked as dependents of it.
+   */
+  get status() {
+    // Unconditionally read the status signal to ensure that any signals that
+    // read it are tracked as dependents.
+    const currentState = this.#status.get();
+    return this.#isNotified ? "pending" : currentState;
+  }
+
+  /**
+   * The last value that the compute function resolved with, or `undefined` if
+   * the last run of the compute function threw an error, or if the compute
+   * function has not yet been run.
+   *
+   * This value is read from a signal, so any signals that read it will be
+   * tracked as dependents of it.
+   */
+  #value: Signal.State<T | undefined>;
+  get value() {
+    this.run();
+    return this.#value.get();
+  }
+
+  /**
+   * The last error that the compute function threw, or `undefined` if the last
+   * run of the compute function resolved successfully, or if the compute
+   * function has not yet been run.
+   *
+   * This value is read from a signal, so any signals that read it will be
+   * tracked as dependents of it.
+   */
+  #error = new Signal.State<unknown | undefined>(undefined);
+  get error() {
+    this.run();
+    return this.#error.get();
+  }
+
+  #deferred = new Signal.State<PromiseWithResolvers<T> | undefined>(undefined);
+
+  /**
+   * A promise that resolves when the compute function has completed, or rejects
+   * if the compute function throws an error.
+   *
+   * If a new run of the compute function is started before the previous run has
+   * completed, the promise will resolve with the result of the new run.
+   */
+  get complete(): Promise<T> {
+    this.run();
+    // run() will have created a new deferred if needed.
+    return this.#deferred.get()!.promise;
+  }
+
+  #computed: Signal.Computed<void>;
+
+  #watcher: Signal.subtle.Watcher;
+
+  // A unique ID for the current run. This is used to ensure that runs that have
+  // been preempted by a new run do not update state or resolve the deferred
+  // with the wrong result.
+  #currentRunId = 0;
+
+  #currentAbortController?: AbortController;
+
+  /**
+   * Creates a new AsyncComputed signal.
+   *
+   * @param fn The function that performs the asynchronous computation. Any
```

Comment:
While there can't be fully automatic tracking for any async signal reads in the computation, it could be supported via an API. Consider:

`fn: (abort: AbortSignal, track<K>: (() => K) => K) => Promise<T>`. 

The `track` function would simply pass through the returned value but internally it would do so via a computed which could then be watched by the watcher to trigger a re-run.